server/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreatePartitions.java

```diff
@@ -275,7 +235,18 @@ public ClusterState executeCreateIndices(ClusterState currentState, CreatePartit
 
             // "Probe" creation of the first index passed validation. Now add all indices to the cluster state metadata and update routing.
 
-            MappingMetadata mappingMetadata = new MappingMetadata(template.mapping());
+            final MappingMetadata mapping = new MappingMetadata(Map.of("default", MappingUtil.createMapping(
```

Comment:
I suspect this is also follow up: See if we can remove the mapping as well - and re-create it on demand based on the RelationMetadata.Table for the BWC cases similar to the templates?

Reply:
Yes, removing the mapping has a lot of impacts (e.g. IndexService.validateMapping()), so I planned to work on this as a followup.